web/src/Storage.jsx

```diff
@@ -31,47 +33,57 @@ export default function Storage() {
   const [state, dispatch] = useReducer(reducer, {
     targets: [],
     target: "",
-    proposal: []
+    actions: [],
+    error: false
   });
-  const { target, targets, proposal } = state;
+  const { target, targets, actions, error } = state;
 
   const onAccept = selected =>
     client
-      .setOption("Disk", selected)
-      .then(() => dispatch({ type: "CHANGE_TARGET", payload: selected }));
+      .calculateStorageProposal({ candidateDevices: [selected] })
+      .then(result => {
+        const payload = { selected, error: (result !== 0) };
+        dispatch({ type: "CHANGE_TARGET", payload });
+      });
 
   useEffect(async () => {
-    const proposal = await client.getStorage();
-    const disk = await client.getOption("Disk");
-    const disks = await client.getDisks();
+    const { availableDevices: disks, candidateDevices: [disk] } = await client.getStorageProposal();
+    const actions = await client.getStorageActions();
     dispatch({
       type: "LOAD",
-      payload: { target: disk, targets: disks, proposal }
+      payload: { target: disk, targets: disks, actions }
     });
   }, []);
 
   useEffect(() => {
     // TODO: abstract D-Bus details
     return client.onPropertyChanged((_path, _iface, _signal, args) => {
-      const [_, changes] = args;
-      if (Object.keys(changes).includes("Disk")) {
-        client
-          .getStorage()
-          .then(proposal =>
-            dispatch({ type: "UPDATE_PROPOSAL", payload: proposal })
-          );
+      const [iface, properties] = args;
+
+      if (iface !== "org.opensuse.DInstaller.Storage.Actions1") {
```

Comment:
Maybe we could have a JS module maping ifaces and exporting them as constants. I mean, a _ifaces.js_ file or so

```js
const ACTIONS = "org.opensuse.DInstaller.Storage.Actions1";
const  WHATERVER = "org.opensuse.DInstaller.Storage.Actions1";

...

export {
  ACTIONS,
  WHATEVER,
  ...
};
```

Then, 

```js

import { ACTIONS as ACTIONS_IFACE } from 'ifaces';

...

if (iface !== ACTIONS_IFACE) {
 
...

```

Reply:
The idea is that this callback does not expose D-Bus internals (so you should not get the interface name and so on). Let's wait until we refactor `InstallerClient`.